src/pages/settings/Profile/Contacts/NewContactMethodPage.js

```diff
@@ -55,84 +51,72 @@ const defaultProps = {
 };
 
 function NewContactMethodPage(props) {
-    const [login, setLogin] = useState('');
-    const [password, setPassword] = useState('');
-    const loginInputRef = useRef(null);
-
-    const handleLoginChange = useCallback((value) => {
-        setLogin(value.trim());
-    }, []);
-
-    const handlePasswordChange = useCallback((value) => {
-        setPassword(value.trim());
-    }, []);
-
-    const isFormValid = useMemo(() => {
-        const phoneLogin = LoginUtils.getPhoneNumberWithoutSpecialChars(login);
-
-        return (Permissions.canUsePasswordlessLogins(props.betas) || password)
-            && (Str.isValidEmail(login) || Str.isValidPhone(phoneLogin));
-    }, [login, password, props.betas]);
-
-    const submitForm = useCallback(() => {
-        // If this login already exists, just go back.
-        if (lodashGet(props.loginList, login)) {
-            Navigation.navigate(ROUTES.SETTINGS_CONTACT_METHODS);
-            return;
+    const isFormValid = (values) => {
+        const phoneLogin = !_.isEmpty(values.phoneOrEmail) ? LoginUtils.getPhoneNumberWithoutSpecialChars(values.phoneOrEmail) : '';
+
+        const errors = {};
+
+        if (_.isEmpty(values.phoneOrEmail)) {
+            errors.phoneOrEmail = props.translate('contacts.genericFailureMessages.contactMethodRequired');
+        }
+
+        if (!_.isEmpty(values.phoneOrEmail) && !(Str.isValidPhone(phoneLogin) || Str.isValidEmail(values.phoneOrEmail))) {
+            errors.phoneOrEmail = props.translate('contacts.genericFailureMessages.invalidContactMethod');
+        }
+
+        if (!_.isEmpty(values.phoneOrEmail) && lodashGet(props.loginList, (values.phoneOrEmail).toLowerCase())) {
+            errors.phoneOrEmail = props.translate('contacts.genericFailureMessages.enteredMethodIsAlreadySubmited');
+        }
+
+        if (!Permissions.canUsePasswordlessLogins(props.betas) || _.isEmpty(values.password)) {
```

Comment:
@dhairyasenjaliya Is there any reason you keep validating the `password` even though the account is passwordless?

Reply:
How about `!Permissions.canUsePasswordlessLogins(props.betas) && _.isEmpty(values.password)`?